test/integration/http2_flood_integration_test.cc

```diff
@@ -216,9 +216,9 @@ void Http2FloodMitigationTest::floodClient(const Http2Frame& frame, uint32_t num
 
   // Upstream connection should be disconnected
   ASSERT_TRUE(fake_upstream_connection_->waitForDisconnect());
-  // Downstream client should receive 503 since upstream did not send response headers yet
+  // Downstream client should receive 502 since upstream did not send response headers yet
```

Comment:
Let me check why codes treats flooding it as protocol error.

Reply:
AFIK it's because this PR assumes codec errors are protocol errors.  *almost* all of them are, but flooding is an example exception where the codec is guarding Envoy against other attack vectors

Reply:
The goal of this PR was to clearly distinguish incorrect responses from upstream and mark them with 502 code. Flooding seems to fall into the same bucket and I think it would be better to respond at least with 503 (as it was before) or, as you suggest, with 500. Let me dig deeper into the codec.

Reply:
I checked the flood tests and had to replace 503 error code with 502 only for tests where Envoy is flooded with incorrect frames from upstream. The rest of flood tests return 503.